src/ds3.c

```diff
@@ -754,6 +754,43 @@ static uint64_t xml_get_bool_from_attribute(const ds3_log* log, xmlDocPtr doc, s
     return xml_get_bool(log, doc, (xmlNodePtr) attribute);
 }
 
+static ds3_error* _get_request_xml_nodes(
+        const ds3_client* client,
+        const ds3_request* request,
+        xmlDocPtr* doc,
+        xmlNodePtr* root,
+        char* root_element_name) {
+    GByteArray* xml_blob = g_byte_array_new();
+
+    ds3_error* error = _internal_request_dispatcher(client, request, xml_blob, ds3_load_buffer, NULL, NULL);
+    if (error != NULL) {
+        g_byte_array_free(xml_blob, TRUE);
+        return error;
+    }
+
+    *doc = xmlParseMemory((const char*) xml_blob->data, xml_blob->len);
```

Comment:
obfuscation.... working through refactoring that but I'm hitting a couple of issues